src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const act in actions) {
+    const type = actions[act].type;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = actions[act];
+
+        Object.assign(copyState, extraData);
+
+        result.push({ ...copyState });
+        break;
+      case 'removeProperties':
+        const { keysToRemove } = actions[act];
+
+        keysToRemove.forEach(key => delete copyState[key]);
+        result.push({ ...copyState });
+        break;
+      default :
+        Object.keys(copyState).forEach(key => delete copyState[key]);
+
+        result.push({ ...copyState });
```

Comment:
you duplicate it for three times, let us think how can we write this line only once